draco/analysis/flagging.py

```diff
@@ -1121,8 +1131,11 @@ def process(self, tstream, rfimask):
         else:  # self.share == "none"
             tsc = tstream.copy()
 
-        # Mask the data
-        tsc.weight[:] *= (~rfimask.mask[sf:ef][bcast_slice]).astype(np.float32)
+        # Mask the data. This will throw an mpiarray warning because the mask being
+        # applied is a numpy array matching the local array directly
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            tsc.weight[:] *= (~rfimask.mask[sf:ef][bcast_slice]).astype(np.float32)
```

Comment:
I'm on the fence about this one. It might be better to create an MPIArray with the mask and then apply directly. Turning off warnings seems a bit heavy handed as maybe some other part of the code could throw a warning in here.

Reply:
Maybe an optional argument to cast the mask to an MPIarray? I agree that it might be a bit much to ignore the warning

Reply:
It could be reasonable to add a method to `MemDatasetCommon` which returns the dataset as an MPIArray distributed along a given axis. Or a classmethod on `MPIArray` which given a common numpy array, builds a distributed MPIArray from it.

Reply:
Is there a specific reason behind not making the draco RFImask distributed?

Reply:
I think it's too small to make it worth the overhead. I agree that on reflection it doesn't seem like the smartest idea, we could consider changing it but I think we need to figure out exactly where it could cause problems downstream first.

Reply:
It might be worth just letting the warning get logged and leaving it as-is. Since `MemDatasetDistributed` exists, it doesn't really make sense to add anything to it related to mpi. This comparison of an mpiarray to a np.ndarray is the intended behaviour here, and it should always work, so maybe it isn't worth trying to hack around it